.github/workflows/inspektor-gadget.yml

```diff
@@ -1342,6 +1342,55 @@ jobs:
           BUILDER_IMAGE=${{ needs.build-helper-images.outputs.ebpf_builder_image }} \
           push-gadgets -o install/ig
 
+  test-gadgets:
+    name: Test gadgets
+    #level 2
+    needs:
+      - build-ig
+      - build-helper-images
+    runs-on: ubuntu-latest
+    permissions:
+      # allow publishing container image
+      # in case of public fork repo/packages permissions will always be read
+      contents: read
+      packages: write
+    steps:
+    - uses: actions/checkout@v4
+    - name: Login to Container Registry
```

Comment:
If we don't plan to push gadgets we shouldn't be logging in? CI might fail because of that.

Reply:
CI was failing till we provided this step: 
https://github.com/pawarpranav83/inspektor-gadget/actions/runs/8567763465/job/23480415618.
And after rebasing the read and write permissions were required to pass the test.